### PR TITLE
Improve update logging in Mongo data manager

### PR DESCRIPTION
**Pull request by peiyaoh**
Adds a warning log when an update operation matches an item but does not modify it. The server will still return the document, instead of null (so won't be an error). Refactors conditional logic for clarity in updateItemInCollection.

### Description

-

### Related Jira Tickets

-

### Comments 

-

Please ensure all items are complete before requesting a review:

- [ ] **Code Completed**: All code for the feature or bugfix has been implemented.
- [ ] **Unit Tests Added**: Unit tests are written for all new functionality, covering edge cases where applicable.
- [ ] **Integration Tests (where necessary)**: Integration tests are added for interactions between modules or components.
- [ ] **All Tests Passing**: All unit and integration tests are passing.
- [ ] **Type Safety Checked**: Ensure TypeScript types are correctly defined and used. Avoid `any` types without proper justification.
- [ ] **Typedoc Comments Added**: Ensure all functions, classes, interfaces, and types are documented using TypeDoc syntax.
- [ ] **Code Formatting**: All code files are formatted according to the project’s standards (e.g., Prettier, ESLint) and no linting errors remain.
- [ ] **No Console Logs**: Remove any `console.log` or other debugging output from the code.
- [ ] **Documentation Updated**: Update relevant project documentation (e.g., README, configuration guides) if necessary.

---

### CODE REVIEW

**Review by numerator — DISMISSED** [review]

File: src/data-manager/mongo/mongo-data-manager.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ const getCollectionChangeReadable = (
     .watch(filterList, options);
 
   const collectionChangeReadable = new Readable({ objectMode: true });
-  collectionChangeReadable._read = () => {};
+  collectionChangeReadable._read = () => { };
```
Comment (jadrubio) [inline src/data-manager/mongo/mongo-data-manager.ts]: no need for space

File: src/data-manager/mongo/mongo-data-manager.ts [hunk 2]
```diff
@@ -220,12 +220,17 @@ const updateItemInCollection = async (
       .collection(collectionName)
       .updateOne({ _id: objectId }, { $set: updateObject });
 
-    if (matchedCount === 1 && modifiedCount === 1) {
+    if (matchedCount === 1) {
+      if (modifiedCount == 0) {
+        Log.warn(`No changes made for item with id ${id} in ${collectionName}`);
+      }
```
Comment (jadrubio) [inline src/data-manager/mongo/mongo-data-manager.ts:226]: it seems that if matchedCount === 1 and modifiedCount === 0  we log the warning but still perform the operation.  is this the intended idea?

Comment (peiyaoh) [reply src/data-manager/mongo/mongo-data-manager.ts:226]: The intended idea is not to throw errors when the document already has the identical values as the updates, which will result in modifiedCount == 0; 

As we discussed in our meeting, an alternative could be added in the future to update the "updatedAt" properties of a document so that the modifiedCount will still be 1.

Comment (jadrubio) [reply src/data-manager/mongo/mongo-data-manager.ts:226]: Yes I rememeber that.  I am not saying we should throw an error merely wondering if we still need to perfom the call to the db to update

Comment (peiyaoh) [reply src/data-manager/mongo/mongo-data-manager.ts:226]: I see. The update should have already been attempted by this point. If we want to prevent the update db update operation, we will first have to check whether the values are the same already, either in the mongo manager or the database manager. Is that what you have in mind?

**Review by jadrubio — APPROVED** [review]
